### DISCUSSION TITLE
How do you fix indenting?

### DISCUSSION THREAD

**Discussion by adam-nielsen** [PHPCSStandards/PHP_CodeSniffer#399] [Q&A]
I have a bunch of poorly formatted code, but I can't figure out how to get basic stuff like the indenting fixed.  For example this file:

```php
if (!$a) {
    if ($b) {
                    // Single line comment indented too far.
                     $a = $b;
    } 
else {
                   /* Comment indented too far. */
                 $x = true;

          /* Another comment indented incorrectly. */
                       $y = true;
    }
}
```

When I run it, it says:
```
$ php phpcbf.phar --standard=PSR12 test.php 

PHPCBF RESULT SUMMARY
----------------------------------------
FILE                    FIXED  REMAINING
----------------------------------------
test.php                4      0
----------------------------------------
A TOTAL OF 4 ERRORS WERE FIXED IN 1 FILE
----------------------------------------

Time: 65ms; Memory: 8MB
```

But all it has done is put that `else` back on the same line, it hasn't fixed the indenting of any statements or of the comments inside the curly braces.

What am I doing wrong?

**Comment by jrfnl** [comment]
@adam-nielsen I can reproduce the behaviour you are describing. This sounds more like something which should be filed as a bug report.

**Reply by adam-nielsen** [reply]
Thanks for your response!  I have created #405 for this.

**Comment by stronk7** [comment]
For some reason it sounds to me (from the old days), that the sniff, by default only reports missing spaces, and not exceeding ones. And there is some sort of “strict” property to look for both.

I don’t remember the details, but that may explain the behaviour… will try to look which exact Sniff I’m talking about.

**Reply by stronk7** [reply]
Yep, it’s this: https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/src/Standards/Generic/Sniffs/WhiteSpace/ScopeIndentSniff.php#L45

And the docs: https://github.com/squizlabs/PHP_CodeSniffer/wiki/Customisable-Sniff-Properties#genericwhitespacescopeindent

**Reply by adam-nielsen** [reply]
That was indeed the problem!  Manually enabling the rule makes it work the way I'm used to with linters in other languages.  Thanks for your advice!
